observability/src/main/java/io/grpc/protobuf/observability/LogRecordExtension.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.protobuf.observability;
+
+import io.grpc.observabilitylog.v1.GrpcLogRecord;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+
+
+public class LogRecordExtension extends LogRecord {
```

Comment:
- mark this class `final`?
- does it need to be `public` ? May be it does if it is going to be accessed from outside the package but we can discuss
- Javadoc for the class
- based on public or not mark it as [`@Internal`](https://github.com/grpc/grpc-java/blob/master/api/src/main/java/io/grpc/Internal.java)

Reply:
Marked it as final and @Internal. Also added javadoc for the class.